### ISSUE TITLE
Increasing texture size of decals crashes the client.

### ISSUE THREAD

**Issue by GringoRSA** [anegostudios/VintageStory-Issues#284]
**Increasing texture size of decals crashes the client with the following crash log:**

```
Version: v1.9.2 (Stable)2019/05/08 19:37:22: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at _ueWd7V39GxoG0N1vvuvuxzXsfRM._OwFp9kXl7KJfJGuabkGdn112TRp(_GCsXMZo9Oac9jP7h0qWBu3BuncB )
   at _ueWd7V39GxoG0N1vvuvuxzXsfRM._8FgBRGqxZs8RZgLkfKhu6ECzRZM(_nQVkny4lXTN8b8MKsPJyKF4XLqc )
   at _nD9jN8XAzh69Y0FFUob7GRf1vke._rWTPQ3wXaxvjySoVFRhzmEv41h(_nQVkny4lXTN8b8MKsPJyKF4XLqc )
   at _DHhjZJnIUs1on6vpFNQcMpMEVjK._lUIyzbqh5jM9WROcAkKikp25BFK(BlockSelection , Block , Single )
   at _DHhjZJnIUs1on6vpFNQcMpMEVjK._gTFZZVxbSNi8kwDfyasq7kj0ngx(Single )
   at _DHhjZJnIUs1on6vpFNQcMpMEVjK._YLEbUy3tF8EXHvUTOqHb2kwxGTw(Single )
   at _nD9jN8XAzh69Y0FFUob7GRf1vke._6bb2EfvFcJy1706Dl4fLaQJoeX(EnumRenderStage , Single )
   at _4tEiYgqX9Tbp0u6XQgZRQN2C00j._6bb2EfvFcJy1706Dl4fLaQJoeX(EnumRenderStage , Single )
   at _4tEiYgqX9Tbp0u6XQgZRQN2C00j._VX1QbjWJlmLJSVlUtjvd8V08RfE(Single )
   at _goFz1wnqfWJkTiNQhkMnFLwovnD._VX1QbjWJlmLJSVlUtjvd8V08RfE(Single )
   at _qMOZ0nqoV8D9qK6wSKJdms05DFe._OdjmEOIogayAlqcQz2BA7nx7rdy(Single )
   at _qMOZ0nqoV8D9qK6wSKJdms05DFe._DkcD3GjMcE3kO8Lf23ZEl9l0FvC(Single )
   at _Q4GkhL8fdNnbyajnFduEGWeDb1f._QoMfbzHHgFiFq1oQkgreZCSLDQX(Object , FrameEventArgs )
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _i2ESdD37oIQUDTNBET1OO6KKA5P._sEL4TL65ecQmjpf9cyH76MTBAuO(_PeiLuq7npShFRqWQoaqgQ2aAB7L , String[] )
   at _xzOEtcS097gqlSlz75yjjYiA28I._sEL4TL65ecQmjpf9cyH76MTBAuO(ThreadStart )
```

**Comment by tyronx** [comment]
Can you send me that texture pack? I to resized my decals but it did not crash for me

**Comment by GringoRSA** [comment]
Here is a link to the texture pack. Not all textures have been replaced yet.

https://drive.google.com/open?id=1SRV8O_ky4Aw3KGWzNVf4ljbnMn-_6kZX

**Comment by tenabraex** [comment]
I commented in the sub as well, but I wonder if its related to the texture atlas issue we had when I made the weapon mod... which I think was related to non-square overlays or overlays of a different size to the base image ?

**Comment by tyronx** [comment]
Hokay! Now I made the whole high res texture pack thing not crashy and prettier too.  

1. The problem was the decal texture atlas was too small to hold the 128x128 decal textures. Now it autmatically adjusts its size. 
2. To tell the game what the texture packs texture resolution is, I added a new propert "textureSize" to the modinfo.json which also fixes #285 where users would need to manually change the textureSize value in the clientsettings.json

Will be in v1.9.4

One problem still remains though, on high res textures packs the block texture atlas will overflow and while it does create a second texture atlas, the engine does not yet properly handle that second one yet and thus some blocks will have the wrong texture. This does need to be fixed manually or by a code mod by setting the maxTextureAtlasSize to 4096 or higher. This is however belongs to another github issue entry.
